### DISCUSSION TITLE
Shortcut to instantiate classes.

### DISCUSSION THREAD

**Discussion by proofit404** [hynek/svcs#114] [Ideas]
Let's say we have a service with high level business logic:
```python
from attrs import define

@define
class Subscription:
    load_order: LoadOrder
    load_customer: LoadCustomer
    create_payment: CreatePayment

    def purchase(self, state):
        ...
```
To get an instance of this service I would have to define custom factory and manually take all constructor arguments by type from containers.
```python
registry = svcs.Registry()

def subscription_factory(svcs_container):
    return Subscription(
        svcs_container.get(LoadOrder),
        svcs_container.get(LoadCustomer),
        svcs_container.get(CreatePayment),
    )

registry.register_factory(Subscription, subscription_factory)
```
This is repetitive and requires to change factory every time I change services class init signature.

I would suggest either
* add `registry.register_class(Subscription)` to be similar to `registry.register_value(str, "Hello World!")`
* add `initialize` helper `registry.register_factory(Subscription, initialize(Subscription))`

Thoughts?

**Comment by hynek** [comment]
@proofit404!!! Good to see you!

Your suggestion is basically https://github.com/hynek/svcs/discussions/94, no? 😇

**Reply by proofit404** [reply]
@hynek good to see you, my friend!

It took me a while to accept my new reality.
